proposals/NNNN-add-preferred-FPS.md

```diff
@@ -0,0 +1,128 @@
+        weakSelf.videoEncoderSettings[(__bridge NSString *)kVTCompressionPropertyKey_ExpectedFrameRate] = @(preferredFPS);
+        ...
+    } else {
+        ...
+        weakSelf.videoEncoderSettings[(__bridge NSString *)kVTCompressionPropertyKey_ExpectedFrameRate] = @(defaultFrameRate);
+    }
+
+}
+```
+Note that videoEncoderSettings property must be NSMutableDictionary as it now would be mutable.
+
+### iOS mobile proxy consideration
+
+A frame rate of iOS SDL application highly depends on how the application captures the video data.
+
+If an application uses combination of SDLCarWindow and SDLStreamingVideoLifecycleManager, above change should take account of
+1) videoEncoderSettings used in SDLStreamingVideoLifecycleManager,
+2) capture rate of CADisplayLink#preferredFramePerSecond (available in iOS10 or higher)
+
+So that should suffice the needs.
+
+If an application uses AVCaptureDevice directly, and does not rely on SDLStreamingVideoLifecycleManager at all, the frame rate must fit into AVFrameRateRange in ACCaptureDevice.format.
+
+Regarding kVTCompressionPropertyKey_ExpectedFrameRate in videoEncoderSettings, however, [iOS documentation](https://developer.apple.com/documentation/videotoolbox/kvtcompressionpropertykey_expectedframerate) says, “provided as a hint to the video encoder” and “The actual frame rate depends on frame durations and may vary.”
+Therefore, developers must make sure taking account of preferredFPS value for capturing video frames (especially if the app does not use SDLCarWindow), as well as configuring videoEncoderSettings.
+
+### Android mobile proxy consideration
+
+Android Java Suite library introduces the combination of VirtualDisplay and MediaEncoder to capture a SdlRemoteDisplay class and encode the video stream.
+When encoding the video frame, a frame rate can be specified by MediaCodec#configure, but [Android documentation](https://developer.android.com/reference/android/media/MediaFormat.html#KEY_FRAME_RATE) says,
+
+"For video encoders this value corresponds to the intended frame rate, although encoders are expected to support variable frame rate based on MediaCodec.BufferInfo#presentationTimeUs".
+
+Therefore situation would be the same as iOS; i.e. mobile developer must take care of actual frame rate.
+
+The actual frame rate of MediaEncoder depends on the frequency of input surface's update in VirtualDisplay, and varies on the application's (SdlRemoteDisplay's) content.
+
+For instance, if the VirtualDisplay gets updated aggressively, the frame rate would be larger than 60 fps, while in idle time, the frame rate will be something like 20 fps.
+
+This issue will be addressed by another proposal later.
+
+## Potential Downsides
+
+Adding an optional parameter to VideoStreamingCapability has no downside. It will be specified by head unit, but how it is used is still up to mobile application.

```

Comment:
```suggestion
Adding an optional parameter to `VideoStreamingCapability` has no downside. It will be specified by head unit, but how it is used is still up to the mobile application.
```